File: sponge-standalone/examples/standalone/error_reporting_trigger/error_reporting_trigger.py
"""
Sponge Knowledge base
Error reporting
"""

class HelloWorldTrigger(Trigger):
    def onConfigure(self):
        self.event = "helloEvent"
    def onRun(self, event):
        whatIsThis.doSomething()
        print event.get("say")

def onStartup():
    sponge.event("helloEvent").set("say", "Hello World!").send()

File: sponge-standalone/examples/standalone/error_reporting_trigger/error_reporting_trigger.js
/**
 * Sponge Knowledge base 
 * Error reporting
 */

var HelloWorldTrigger = Java.extend(Trigger, {
    onConfigure: function(self) {
        self.withEvent("helloEvent");
    },
    onRun : function(self, event) {
        whatIsThis.doSomething();
        print(event.get("say"));
    }
});

function onStartup() {
    sponge.event("helloEvent").set("say", "Hello World!").send();
}
